tests/utapi/utilities.js

```diff
@@ -0,0 +1,471 @@
+        outgoingBytes: 0,
+    },
+}, {
+    action: 'putBucketNotification',
+    metrics: {
+        bucket: 'notificationtestbucket',
+    },
+    expected: {
+        objectDelta: undefined,
+        sizeDelta: undefined,
+        incomingBytes: undefined,
+        outgoingBytes: 0,
+    },
+}, {
+    action: 'putBucketObjectLock',
+    metrics: {
+        bucket: 'mock-bucket',
+    },
+    expected: {
+        objectDelta: undefined,
+        sizeDelta: undefined,
+        incomingBytes: undefined,
+        outgoingBytes: 0,
+    },
+}, {
+    action: 'putBucketReplication',
+    metrics: {
+        bucket: 'source-bucket',
+    },
+    expected: {
+        objectDelta: undefined,
+        sizeDelta: undefined,
+        incomingBytes: undefined,
+        outgoingBytes: 0,
+    },
+}, {
+    action: 'putBucketVersioning',
+    metrics: {
+        bucket: 'source-bucket',
+    },
+    expected: {
+        objectDelta: undefined,
+        sizeDelta: undefined,
+        incomingBytes: undefined,
+        outgoingBytes: 0,
+    },
+}, {
+    action: 'putBucketWebsite',
+    metrics: {
+        bucket: 'testgetwebsitetestbucket',
+    },
+    expected: {
+        objectDelta: undefined,
+        sizeDelta: undefined,
+        incomingBytes: undefined,
+        outgoingBytes: 0,
+    },
+}, {
+    action: 'uploadPart',
+    metrics: {
+        bucket: 'ftest-mybucket-74',
+        keys: ['toAbort&<>"\''],
+        newByteLength: 5242880,
+        oldByteLength: null,
+        location: 'us-east-1',
+    },
+    expected: {
+        objectDelta: undefined,
+        sizeDelta: 5242880,
+        incomingBytes: 5242880,
+        outgoingBytes: 0,
+    },
+}, {
+    action: 'uploadPartCopy',
+    metrics: {
+        bucket: 'destinationbucket815502017',
+        keys: ['copycatobject'],
+        newByteLength: 26,
+        oldByteLength: null,
+        location: 'us-east-1',
+    },
+    expected: {
+        objectDelta: undefined,
+        sizeDelta: 26,
+        incomingBytes: 26,
+        outgoingBytes: 0,
+    },
+}];
+
+describe('utapi v2 pushmetrics utility', () => {
+    const log = new werelogs.Logger('utapi-utility');
+    const { UtapiClient, utapiVersion } = require('utapi');
+    let pushMetric;
+
+    function pushMetricStub(event) {
+        return event;
+    }
+
+    before(() => {
+        assert.strictEqual(utapiVersion, 2);
+        sinon.stub(UtapiClient.prototype, 'pushMetric')
+            .callsFake(pushMetricStub);
+        pushMetric = require('../../lib/utapi/utilities').pushMetric;
+    });
+
+    after(() => {
+        sinon.restore();
+    });
+
+    testEvents.forEach(event => {
+        it(`should compute and push metrics for ${event.action}`, () => {
+            const eventPushed = pushMetric(event.action, log, event.metrics);
+            assert(eventPushed);
+            Object.keys(event.expected).forEach(key => {
+                assert.deepStrictEqual(eventPushed[key], event.expected[key]);
+            });
```

Comment:
I think just one assertion will work here.
```suggestion
            assert.deepStrictEqual(eventPushed, event.expected);
```

Reply:
Not all properties of the published event are compared, only metrics.